.github/workflows/label_pr_approval.yaml

```
@@ -0,0 +1,53 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Provide approval for bot PRs
+
+on:
+  workflow_call:
+    inputs:
+      labels:
+        description: List of labels in JSON format, i.e. '["trivial", "documentation"]', to indicate that a PR should be approved. When a pull request contains at least one of those labels, the PR will be approved.
+        type: string
+        default: '["trivial"]'
+
+jobs:
+  approve_pr:
+    name: Approve PR for matching label
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
```

```suggestion
    if: github.event_name == 'pull_request' && github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name
```

I think we should probably disable this workflow for pull requests initiated from forks, as running it from a fork may fail.